db-compare-server/api.go

```diff
@@ -4,94 +4,80 @@ import (
 	"encoding/json"
 	"fmt"
 	"net/http"
+
+	"github.com/umamimike/db-compare/db-compare-server/db"
 )
 
-//Convenience for typing
+var storageDir = "app-db-data/"
+
 type msi = map[string]interface{}
 
-func indexHandler(w http.ResponseWriter, r *http.Request) {
+//Convenience for typing
 
+func indexHandler(w http.ResponseWriter, r *http.Request) {
 	response := msi{
 		"data": "stub",
 	}
+	json.NewEncoder(w).Encode(response)
 
-	jsonData, err := json.MarshalIndent(response, "", "  ")
-	if err != nil {
-		fmt.Println(err)
-	}
-	setJsonHeaders(w)
-	w.Write(jsonData)
+}
+
+// ------------- Wrapper structs -----------------
 
+type Data struct {
+	Id         string `json:"id"`
+	Type       string `json:"type"`
+	Attributes msi    `json:"attributes"`
 }
 
-type (
-	JsonApi struct {
-		Data Data `json:"data"`
-	}
-	Data struct {
-		Id         string `json:"id"`
-		Type       string `json:"type"`
-		Attributes msi    `json:"attributes"`
-	}
+// ------------- datasource --------------------
 
-	DatasourceCredentials struct {
-		Username string `json:"username"`
-		Hostname string `json:"host"`
-		Password string `json:"password"`
-		DbName   string `json:"dbname"`
-	}
-)
+func datasourcesPostHandler(w http.ResponseWriter, r *http.Request) {
 
-func datasourcesHandler(w http.ResponseWriter, r *http.Request) {
+	storage := db.NewBadger(storageDir)
+	defer storage.Close()
 
 	var dsr DatasourceCredentials
 	decoder := json.NewDecoder(r.Body)
 	decoder.Decode(&dsr)
 
+	ID, _ := newUUID()
 	// passing the input data back to the output
-	response := JsonApi{
-		Data: Data{
-			Type: "datasource",
-			Id:   "1",
-			Attributes: msi{
-				"username": dsr.Username,
-				"host":     dsr.Hostname,
-				"password": dsr.Password,
-				"dbname":   dsr.DbName,
-			},
+	resp := &Data{
+		Id:   ID,
+		Type: "datasource",
+		Attributes: msi{
+			"username": dsr.Username,
+			"hostname": dsr.Hostname,
+			"dbname":   dsr.DbName,
 		},
 	}
 
-	jsonData, err := json.MarshalIndent(response, "", "  ")
-	if err != nil {
-		fmt.Println(err)
-	}
-	setJsonApiHeaders(w)
-	w.Write(jsonData)
+	// storage.Set(ID, response)
+	// setJsonApiHeaders(w)
```

Comment:
let's not leave commented code in a commit. delete it! if you need to see this later, you can always use git to look at old code.